tests/validator.py

```diff
@@ -0,0 +1,121 @@
+import unittest
+import requests
+
+
+class ValidatorServerTestCase(unittest.TestCase):
+    """
+    This class contains integration tests for the Validator server.
+    """
+
+    def setUp(self):
+        self.base_url = "http://localhost:8000"
+
+    def test_server_running(self):
+        """
+        Test that the server is running and can receive calls through /healthcheck endpoint.
+        """
+        healthcheck_url = f"{self.base_url}/healthcheck"
+        response = requests.get(healthcheck_url)
+        self.assertEqual(
+            response.status_code,
+            200,
+            "Server is not running or not reachable on /healthcheck endpoint",
+        )
+
+    def get_recent_tweets(self):
+        """
+        Test that the server is running and can receive calls through /healthcheck endpoint.
+        """
+        healthcheck_url = f"{self.base_url}/data/twitter/tweets/recent"
+        response = requests.get(healthcheck_url)
+
+        if response.status_code == 200:
+            tweets = response.json()
+            if len(tweets) > 0:
+                success = True
+            else:
+                success = False
+        else:
+            self.assertTrue(False, f"Response status code: {response.status_code}")
+
+        self.assertTrue(success, "No tweets found in fetching recent tweets")
+
+        self.assertEqual(
```

Comment:
should we also maybe assert here that we get some data as well from the JSON response? e.g. at least 1 tweet should be returned

Reply:
Thats happening!
On line 34 `response.json()` is the list of tweets, if that list's length is greater than 0 `success = True`

then if `success == False` => `No tweets found in fetching recent tweets`

Reply:
ouch sorry, my bad!